From d99850ff390b477403c4bf6ab9fcd0ad943a90e1 Mon Sep 17 00:00:00 2001
From: Sebastian Wick <sebastian.wick@redhat.com>
Date: Wed, 17 Apr 2024 12:10:44 +0200
Subject: [PATCH] portal-impl: Check for default_portal before derefing it

The default_portal pointer can be NULL if there is no default portal set
in the configuration. In that case the default portal is not explicitly
set to "none" and we return FALSE in portal_default_prefers_none.

Fixes: 1394bb2 ("portal-impl: fallback implementation is used if preferred value is none")
Closes: https://github.com/flatpak/xdg-desktop-portal/issues/1330
---
 src/portal-impl.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/portal-impl.c b/src/portal-impl.c
index f2bbda203..8c1fd5353 100644
--- a/src/portal-impl.c
+++ b/src/portal-impl.c
@@ -508,7 +508,8 @@ find_matching_iface_config (const char *interface)
 static gboolean
 portal_default_prefers_none (void)
 {
-  if (config != NULL && g_strv_contains ((const char * const *) config->default_portal->portals, "none"))
+  if (config && config->default_portal &&
+      g_strv_contains ((const char * const *) config->default_portal->portals, "none"))
     {
       g_debug ("Found 'none' in configuration for default");
       return TRUE;
